### PR TITLE
Keylime dedicated user account enhancement proposal

### LINKED ISSUE #23: Use a dedicated user account
# Enhancement Description
- One-line enhancement description (can be used as a release note): dedicated Keylime user account
- Keylime Enhancement Proposal: TBC
- Primary contact (assignee): @axelsimon 
- Enhancement target (which target equals to which milestone):
  - Alpha release target (x.y) TBC
  - Beta release target (x.y) TBC
  - Stable release target (x.y) TBC

### PR DESCRIPTION
**Pull request by axelsimon**
Currently, Keylime runs as root.
This enhancement suggests changing running Keylime (specifically, its different components) under a dedicated user
account to limit the level of damage that could happen should a vulnerability in Keylime be exploited.

Some sections remain to be fleshed out, but the core proposal is explained here.

Closes #23.

### CODE REVIEW

File: 23_dedicated_keylime_user_account.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+Finally, the documentation will have to be updated as well.
+
+### User Stories (optional)
+
+#### Story 1
+As a system administrator in an organisation with strong security rules
+regarding the software that may be run, and how, i would like to use Keylime for
+the benefits it provides, but cannot as only a very limited subset of software
+is whitelisted to run as root on the organisation's hosts, and Keylime is not
```
Comment (mpeters) [inline 23_dedicated_keylime_user_account.md:147]: Should: Change "whitelisted" to something like "allowed"

File: 23_dedicated_keylime_user_account.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+one of them.
+
+Now that Keylime uses its own dedicated user account, this aligns Keylime with my
+organisation's policy and i can deploy Keylime.
+
+#### Story 2
```
Comment (mpeters) [inline 23_dedicated_keylime_user_account.md:153]: We can leave off the "Story 2", I think the first user story is definitely enough

**Review by mpeters — CHANGES_REQUESTED** [review]
Just some minor changes

File: 23_dedicated_keylime_user_account.md (hunk 1 above)
Comment (axelsimon) [reply 23_dedicated_keylime_user_account.md:147]: good point, thanks

**Comment by axelsimon** [comment]
I think this updated version addresses @mpeters' comments. I have re-requested review.

File: 23_dedicated_keylime_user_account.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+## Drawbacks
+
+It makes the overall design of Keylime a bit more complicated. However, we
+believe that benefits provided by improving privilege separation vastly
+outweigh this drawback.
+
+## Alternatives
+
+Privilege separation (and more generally, defense in depth) can be done at
```
Comment (ashcrow) [inline 23_dedicated_keylime_user_account.md:222]: Would containerization be an acceptable alternative as well (assuming SPC isn't required)?

File: 23_dedicated_keylime_user_account.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+vulnerability, and if a bug in a piece of software running as root is exploited,
+the negative consequences to the host system can be dramatic, essentially
+allowing the malicious party exploiting the bug to have full control over the
+system (There are some exceptions to this, which we will not consider
+specifically here as they are system-dependent, such as running software in
+TEEs)).
+
+A simple and well-known defense against this is privilege separation,
+specifically the least privileged model, where each piece of software is given
+the rights necessary to run properlpy, but no more. As such, we want to
```
Comment (mpeters) [inline 23_dedicated_keylime_user_account.md:106]: TYPO: Should be "properly"

File: 23_dedicated_keylime_user_account.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+make sure that, in as many cases as possible, Keylime components do not run as
+the root user but as a specific user, whose rights on a host are more
+specifically scoped out, minimising the risk of damage should an exploitable
+vulnerability be found in Keylime.
+
+### Goals
+
+The goals of the enhancement is to move as many components of Keylime as
+possible from runnig as the root user to running as a dedicated user, called
```
Comment (mpeters) [inline 23_dedicated_keylime_user_account.md:115]: TYPO: Should be "running"

File: 23_dedicated_keylime_user_account.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+may be some work to do on the SElinux labels too. 
+
+Finally, the documentation will have to be updated as well.
+
+### User Stories (optional)
+
+#### Story 1
+
+As a system administrator in an organisation with strong security rules
+regarding the software that may be run, and how, i would like to use Keylime for
```
Comment (mpeters) [inline 23_dedicated_keylime_user_account.md:146]: TYPO: Should be "I" (capital)

**Review by mpeters — CHANGES_REQUESTED** [review]
Just some small typos

File: 23_dedicated_keylime_user_account.md (hunk 6 above)
Comment (axelsimon) [reply 23_dedicated_keylime_user_account.md:146]: I've stopped capitalising my “i”, unless at the beginning of a sentence, but i won't fight anyone on this :)

**Comment by axelsimon** [comment]
> Just some small typos

Thanks, nice catch. Fixed!

**Pull request merged**